TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -34,127 +34,104 @@ class BenchmarkParser extends Parser {
             let curRegexResult = null;
             let curMetricValue = null;
-            let curJavaBuildDate = null;
-            let curJavaVersion = null;
             let isValid = true;
             let curTestData = {};
+            let curBenchmarkName = null;
+            let curJavaBuildDate = null;
+            let curJavaVersion = null;
+            let curProductResource = null;
+            let curBenchmarkVariant = null;
+            let curBenchmarkProduct = null;
+            let curBench = null;
 
-            // Parse benchmark name
-            if ( ( curRegexResult = benchmarkNameRegex.exec( curItr.value ) ) !== null ) {
+            //Parse Benchmark name & Java versions 
+            if ( ( curRegexResult = benchmarkNameRegex.exec( curItr.value ) ) != null) {
                 curBenchmarkName = curRegexResult[1];
             } else {
                 isValid = false;
             }
 
-            // Parse benchmark variant
             if ( ( curRegexResult = benchmarkVariantRegex.exec( curItr.value ) ) !== null ) {
                 curBenchmarkVariant = curRegexResult[1];
             } else {
                 isValid = false;
             }
 
-            if (!BenchmarkMetricRouter[curBenchmarkName]) {
-                isValid = false;
-            } else if (!BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]) {
-                isValid = false;
-            // Benchmark should have at least one metric to parse with
-            } else if (!BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]
-            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
-            || !(Array.isArray(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]))) {
-                isValid = false;
-            }
-
-            // Parse benchmark product
             if ( ( curRegexResult = benchmarkProductRegex.exec( curItr.value ) ) !== null ) {
                 curBenchmarkProduct = curRegexResult[1];
+            } else {
+                isValid = false;
             }
 
-            if ( isValid ) {
-
-                curBenchmarkNamedRouted = BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant];
-
-                // Parse metric values
-                curTestData["metrics"] = [];
-                for ( let i = 0; i < BenchmarkMetric[curBenchmarkNamedRouted]["metrics"].length; i++ ) {
-                    curMetric = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].name;
-                    curSearchString = curItr.value;
-
-                    // metric values will be found within the result of the outer regex
-                    if (BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex !== undefined) {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex.exec( curSearchString ) ) !== null ) {
```

Comment:
Not sure why we changed this. Let's not change stuff unless we have a good reason.